src/lib.rs

```diff
@@ -1,7 +1,38 @@
+extern crate regex;
+
+mod links;
+
+#[derive(Debug)]
+pub struct MessageMeta {
+    pub message: String,
+    pub links: Vec<links::Link>,
+}
+
+pub fn parse<T: ToString>(message: T) -> MessageMeta {
+    let msg = message.to_string();
+    let links = links::parse(msg.clone());
+
+    MessageMeta { message: msg.to_string(), links }
+}
+
 #[cfg(test)]
 mod tests {
+    use super::parse;
+
+    #[test]
+    fn should_parse_message_without_meta() {
+        let meta = parse("test message");
+
+        assert_eq!(meta.message, "test message");
+        assert_eq!(meta.links.len(), 0);
+    }
+
     #[test]
-    fn it_works() {
-        assert_eq!(2 + 2, 4);
+    fn should_parse_message_with_link() {
```

Comment:
brakuje jeszcze pewnie kilku testów na edge case - link na początku, na końcu, tekst będący samym linkiem, tekst z wieloma linkami.

Reply:
Widzę, że tam w samym module `links` są niektóre z tych testów. Tak teraz pomyślałem, że wraz z rozwojem tej biblioteki będziesz chciał wyciągać potencjalnie coraz więcej informacji, a wówczas liczba testów może ogromnie wzrosnąć. 

Może warto użyć [quickcheck](https://github.com/BurntSushi/quickcheck), żeby różne edge case'y generowały się same. Nie używałem tej libki w Rust, ale sam quickcheck pochodzi z Haskella.

Pomysł polega na tym, że tworzysz sobie generatory poprawnych danych wejściowych i określasz jakie mają być właściwości danych wyjściowych. I quickcheck sam generuje tony przykładów i jak coś nie zabangla, to redukuje przykład do najmniejszego możliwego failującego.

Np. jak piszesz funkcję do sortowania tablicy liczb. To generator wejściowy będzie generował po prostu losowe tablice liczb. Na wyjściu sprawdzasz tylko stałą właściwość jakiej oczekujesz, czyli w przypadku posortowanych liczb np.: "każdy kolejny element jest większy od poprzedniego".

Reply:
brzmi to bardzo kusząco, nawet nie wiedziałem że quickcheck jest przeniesiony z haskel-a ale świetny pomysł :)

Zostawiłem sobie to w TODO do tych testów, bo pewnie zajmę się tym oddzielnie, ale pewnie drobnym iprovementem będzie zrobienie testów parametryzowanych na początek :) co ty na to?

Reply:
Pewnie, wprowadzenie quickchecka to zabawa sama w sobie :)